torch/_higher_order_ops/while_loop.py

```diff
@@ -75,14 +75,16 @@ def while_loop(cond_fn, body_fn, carried_inputs):
             return val
 
     Args:
-        cond_fn (Callable): A callable function that returns a boolean Scalar tensor.
+        cond_fn (Callable): A callable function that returns a boolean Scalar tensor or a non-const bool.
```

Comment:
It might be hard for a user to figure out what a "non-const bool" is. This is not exactly a python concept

Reply:
Yeah, agree. We'll just specialize if it's a constant then.